go-controller/pkg/node/node.go

```diff
@@ -652,33 +656,35 @@ func (n *OvnNode) Start(ctx context.Context) error {
 		}
 	}
 
-	// Start the health checking server used by egressip, if EgressIPNodeHealthCheckPort is specified
-	if err := n.startEgressIPHealthCheckingServer(mgmtPortConfig); err != nil {
-		return err
-	}
-
 	klog.Infof("OVN Kube Node initialized and ready.")
 	return nil
 }
 
-func (n *OvnNode) startEgressIPHealthCheckingServer(mgmtPortConfig *managementPortConfig) error {
+func (n *OvnNode) startEgressIPHealthCheckingServer(mgmtPortEntry managementPortEntry) error {
 	healthCheckPort := config.OVNKubernetesFeature.EgressIPNodeHealthCheckPort
 	if healthCheckPort == 0 {
 		klog.Infof("Egress IP health check server skipped: no port specified")
 		return nil
 	}
 
 	var nodeMgmtIP net.IP
-	if mgmtPortConfig.ipv4 != nil {
-		nodeMgmtIP = mgmtPortConfig.ipv4.ifAddr.IP
-	} else if mgmtPortConfig.ipv6 != nil {
-		nodeMgmtIP = mgmtPortConfig.ipv6.ifAddr.IP
-		// Wait for IPv6 address to become usable.
-		if err := ip.SettleAddresses(mgmtPortConfig.ifName, 10); err != nil {
-			return fmt.Errorf("failed start health checking server due to unsettled IPv6: %w", err)
+	var mgmtPortConfig *managementPortConfig = mgmtPortEntry.config
+	// Not all management port interfaces can have IP addresses assignable to them.
+	if mgmtPortEntry.port.HasIpAddr() {
```

Comment:
I think this check should be moved before calling startEgressIPHealthCheckingServer()?

Reply:
I think the check should be done after checking "EgressIPNodeHealthCheckPort" also I think it would be more clear to have all the IP addresses checks in 1 place rather then spreading it out.

Reply:
I hope we are on the same page here.

Reply:
no strong opinions one way or another. but my argument is that we don't need startEgressIPHealthCheckingServer as long as the management port is not the ones with IP addresses, so we should not call that for those mgmtPortEntry from start.